src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -2739,6 +2740,28 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
             Stream pdbStream,
             ICollection<MethodDefinitionHandle> updatedMethods,
             CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var updatedTypes = new List<TypeDefinitionHandle>();
+            return EmitDifference(baseline, edits, isAddedSymbol, metadataStream, ilStream, pdbStream, updatedMethods, updatedTypes, cancellationToken);
+        }
+
+        /// <summary>
+        /// Emit the differences between the compilation and the previous generation
+        /// for Edit and Continue. The differences are expressed as added and changed
+        /// symbols, and are emitted as metadata, IL, and PDB deltas. A representation
+        /// of the current compilation is returned as an EmitBaseline for use in a
+        /// subsequent Edit and Continue.
+        /// </summary>
+        public EmitDifferenceResult EmitDifference(
+            EmitBaseline baseline,
+            IEnumerable<SemanticEdit> edits,
+            Func<ISymbol, bool> isAddedSymbol,
+            Stream metadataStream,
+            Stream ilStream,
+            Stream pdbStream,
+            ICollection<MethodDefinitionHandle> updatedMethods,
+            ICollection<TypeDefinitionHandle> updatedTypes,
```

Comment:
>updatedTypes [](http://example.com/codeflow?start=46&length=12)

I wonder if we should move `updatedMethods` and `updatedTypes` to `EmitDifferenceResult` as `ImmutableArray<MethodDefinitionHandle>` and `ImmutableArray<MethodDefinitionHandle>`, respectively, instead.

They are outputs/results. The previous overload could also populate the collection for backward compat but we can make it [Obsolete].